docs/src/writing-tests-js.md

```diff
@@ -3,90 +3,147 @@ id: writing-tests
+prior to performing the action, so there is no need to need wait for it to become
+available.
+
+
+```js
+// Create a locator.
+const getStarted = page.getByRole('link', { name: 'Get started' });
+// Click it.
+await getStarted.click();
 ```
 
-:::note
-Add `// @ts-check` at the start of each test file when using JavaScript in VS Code to get automatic type checking.
-:::
+In most cases, it'll be written in one line:
 
-### Assertions
+```js
+await page.getByRole('link', { name: 'Get started' }).click();
+```
+
+
+### Basic actions
+
+This is the list of the most popular Playwright actions. Note that there are
+many more, so make sure to check the [Page API](./api/class-page.md) section to
+learn more about them.
+
+| Action | Description |
+| :- | :- |
+| [`method: Locator.check`] | Check the input checkbox |
+| [`method: Locator.click`] | Click the element |
+| [`method: Locator.uncheck`] | Uncheck the input checkbox |
+| [`method: Locator.hover`] | Hover mouse over the element |
+| [`method: Locator.fill`] | Fill the form field (fast) |
+| [`method: Locator.focus`] | Focus the element |
+| [`method: Locator.press`] | Press single key |
+| [`method: Locator.setInputFiles`] | Pick files to upload |
+| [`method: Locator.selectOption`] | Select option in the drop down |
+| [`method: Locator.type`] | Type text character by character (slow) |
+
+## Assertions
 
-Playwright Test uses the [expect](https://jestjs.io/docs/expect) library for [test assertions](./test-assertions.md) which provides matchers like `toEqual`, `toContain`, `toMatch`, `toBe` and many more. Playwright also extends this library with convenience async matchers that will wait until the expected condition is met.
+Playwright Test uses the [expect](https://jestjs.io/docs/expect) library for
+[test assertions](./test-assertions.md) which provides matchers like
+`toEqual`, `toContain`, `toMatch`, `toBe` and many more. It also extends
+this library with the convenience async matchers that will wait until the expected condition is met. Using these matches allows making the tests non-flaky and resilient. For example, this code will wait until
+the page gets the title containing "Playwright":
 
 ```js
 await expect(page).toHaveTitle(/Playwright/);
 ```
 
 
-### Locators
 
-[Locators](./locators.md) are the central piece of Playwright's auto-waiting and retry-ability. Locators represent a way to find element(s) on the page at any moment and are used to perform actions on elements such as `.click` `.fill` etc.
+Here is the list of the most popular async assertions. Note that there are [many more](./test-assertions.md) to get familiar with:
 
-```js
-const getStarted = page.getByRole('link', { name: 'Get started' });
+| Assertion | Description |
+| :- | :- |
+| [expect(locator).toBeChecked](./test-assertions#locator-assertions-to-be-checked) | Checkbox is checked |
```

Comment:
Why not use method: …?

Reply:
Because of no good reason apparently - I mistyped it, it did not work, so I worked around. It does work!